src/hooks/UseForm/useForm/types/public.ts

```diff
@@ -0,0 +1,395 @@
+    ref: React.Ref<TFieldElement>
+  },
+> = TRefProps & {
+  onChange: (event: unknown) => void
+  onFocus: () => void
+  onBlur: () => void
+}
+
+/**
+ * The type of the `register` function provided by `useForm`.
+ * Generic for:
+ * - The form's data type
+ * - The type of element being registered as a field
+ * - Where or not the field is required
+ *
+ * @export
+ * @typedef {RegisterFunction}
+ * @template {FormData} TData
+ */
+export type RegisterFunction<TData extends FormData, TDimensions extends number = 0> = <
+  TFieldName extends keyof TData,
+  TFieldElement extends FieldElement,
+  TIsRequired extends boolean = false,
+>(
+  fieldName: TFieldName,
+  options?: RegisterOptions<
+    TData,
+    TDimensions,
+    TFieldName,
+    keyof RefProps<TFieldElement>,
+    TIsRequired
+  >,
+) => RegisterResult<TFieldElement, RefProps<TFieldElement>>
+
+/**
+ * The callback run when submitting via `handleSubmit` in `useForm`.
+ * If validations are skipped, data is not guaranteed to exist.
+ *
+ * @export
+ * @typedef {OnSubmit}
+ * @template {FormData} TData
+ * @template {boolean} TShouldSkipValidations
+ */
+export type OnSubmit<
+  TData extends FormData,
+  TDimensions extends number = 0,
+  TShouldSkipValidations extends boolean = false,
+> = (
+  data: TShouldSkipValidations extends true
+    ? TDimensions extends 0
+      ? Partial<TData>
+      : IGraph<Partial<TData>, TDimensions>
+    : TDimensions extends 0
+      ? TData
+      : IGraph<TData, TDimensions>,
+) => void
+
+/**
+ * The options for handling a submission with `useForm`
+ *
+ * @export
+ * @interface HandleSubmitOptions
+ * @typedef {HandleSubmitOptions}
+ * @template {FormData} TData
+ * @template {boolean} TShouldSkipValidations
+ */
+export interface HandleSubmitOptions<
+  TData extends FormData,
+  TDimensions extends number = 0,
+  TShouldSkipValidations extends boolean = false,
+> {
+  /**
+   * If `true`, the `onSubmit` function will be called even with errors in the form
+   *
+   * @type {?TShouldSkipValidations}
+   */
+  shouldSkipValidations?: TShouldSkipValidations
+  /**
+   * A callback used upon submission of the form with all data defined
+   * If `shouldSkipValidations` is `true`, data is not guaranteed to exist
+   *
+   * @type {?OnSubmit<TData, TShouldSkipValidations>}
+   */
+  onSubmit?: OnSubmit<TData, TDimensions, TShouldSkipValidations>
+  /**
+   * A callback used upon submission of the form if errors prevented completion
+   *
+   * @type {?(errors: TDimensions extends 0 ? Errors<TData> : IGraph<Errors<TData>, TDimensions>) => void}
+   */
+  onError?: (
+    errors: TDimensions extends 0 ? Errors<TData> : IGraph<Errors<TData>, TDimensions>,
+  ) => void
+}
+
+/**
+ * The type of the `handleSubmit` function provided by `useForm`
+ *
+ * @export
+ * @typedef {HandleSubmit}
+ * @template {FormData} TData
+ * @template {boolean} TShouldSkipValidations
+ */
+export type HandleSubmit<TData extends FormData, TDimensions extends number = 0> = <
```

Comment:
May want to rename this or the `RegisterFunction` type so they're consistent with the `Function` suffix

```suggestion
export type HandleSubmit<TData extends FormData, TDimensions extends number = 0> = <
```